cpp/oneapi/dal/table/backend/homogen_table_impl.cpp

```diff
@@ -30,6 +30,21 @@ table_metadata create_homogen_metadata(std::int64_t feature_count, data_type dty
     return table_metadata{ dtypes, ftypes };
 }
 
+table_metadata create_homogen_metadata(std::int64_t feature_count,
```

Comment:
I'd suggest removing that function and create `dtypes` array explicitly then passing it to the `table_metadata` ctor.